src/java.desktop/share/classes/javax/swing/plaf/synth/SynthGraphicsUtils.java

```diff
@@ -731,4 +744,41 @@ public int getIconHeight() {
             return synthIcon.getIconHeight(context);
         }
     }
+
+    /*
+     * Repaints all the components with the mnemonics in the given window and all its owned windows.
+     */
+    static void repaintMnemonicsInWindow(final Window w) {
```

Comment:
The `repaintMnemonicsInWindow` and `repaintMnemonicsInContainer` are exactly the same as methods in `WindowsGraphicsUtils`. And `AquaMnemonicHandler` has yet another copy of the same code.

Is it possible to move these methods to a utility class that's available for all Look-and-Feels to avoid duplicating code?

Reply:
I will check and update if it is possible.

Reply:
Thank you for looking into it. A `MnemonicHandler` class in `sun.swing` or `sun.swing.plaf` package could be a good candidate. The `sun.swing` package contains a lot of support classes for Swing, including `SwingUtilities2` and `UIAction`; the `sun.swing.plaf` may be better as the mnemonic handler is part of PLAF.

Another option is the `com.sun.java.swing` package which currently contains `SwingUtilities3` and `plaf` subpackage with `gtk` and `motif` related classes.

Reply:
I tried moving the `repaintMnemonicsInWindow` and `repaintMnemonicsInContainer` under `SwingUtilities` class but faced build issue while accessing them in `WindowsPopupMenuUI` file.

I will try with the suggested packages as you mentioned.

Reply:
The `javax.swing.SwingUtilities` is a public class, which means adding public methods to it requires a CSR; package-private methods aren't visible from other packages.

The `sun.swing` and `com.sun.java.swing` packages aren't public, you can add public classes with public methods easily; since the classes are public and the methods are public, they're accessible from any other package in Swing.